tests/query/parser/test_formula_mql_query.py

```diff
@@ -259,7 +279,7 @@ def test_simple_formula() -> None:
             ),
         ],
         groupby=[time_expression],
-        condition=formula_condition,
+        condition=binary_condition("and", metric_id_in({123456}), formula_condition),
```

Comment:
@evanh modifying this file made me think: should the parsing of queries be more distinct than the post processing step.

Reply:
What do you mean by this?

Reply:
here my thinking:
formula mql -> initial ast -> post processed ast

this file tests given a formula mql, is the expected post processed ast generated. this makes it s.t. adding the new post processor requires me to modify these tests.

another design could be: given a formula mql, make sure the correct initial ast is generated
and then: given and initial ast, make sure the correct post processed ast is generated

this isnt possible due to the fact that parsing and post processing are tightly coupled

not saying we should actually do this, just something I thought of when I was modifying the tests

Reply:
no need to reply to this I will bring it up in self serve since its relevant and see if they have any opinions